### PR TITLE
fix to makefile for OSX/Darwin

### PR DESCRIPTION
**Pull request by mandric**

### CODE REVIEW

**Review by drasko — DISMISSED** [review]
LGTM

**Review by dborovcanin — COMMENTED** [review]
Please sign your commits.

**Comment by mandric** [comment]
Sorry, added GPG signature to commit, rebased and pushed again.  Also signed-off on my own commit.

File: Makefile [hunk 1]
```diff
@@ -8,7 +8,7 @@ SERVICES = users things http normalizer ws coap lora influxdb-writer influxdb-re
 DOCKERS = $(addprefix docker_,$(SERVICES))
 DOCKERS_DEV = $(addprefix docker_dev_,$(SERVICES))
 CGO_ENABLED ?= 0
-GOOS ?= linux
+GOOS ?= $(shell go env GOOS || echo linux)
```
Comment (manuio) [inline Makefile]: `GOOS ?= $(go env GOOS)` should also work for linux

Can you please also add this ones:
`GOARCH ?= $(go env GOARCH)`
`GOARM ?= $(go env GOARM)`

**Review by manuio — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (drasko) [reply Makefile]: @mandric please use these simpler checks that @manuio proposed

Comment (mandric) [reply Makefile]: Sure thing! But in my testing it needed `shell` keyword.

Comment (manuio) [reply Makefile]: It's working for us on Linux and OS X without this keyword. Why you need it?

Comment (mandric) [reply Makefile]: Without the shell keyword I'm pretty sure the shell does not get invoked.  It doesn't work for me without shell keyword, I just get an empty string.

Comment (drasko) [reply Makefile]: @mandric might  be that Makefile demands this. Let me re-check.

Comment (drasko) [reply Makefile]: Yes, I checked - `shell` keyword is needed, so please just add it.

Comment (mandric) [reply Makefile]: Another option that works on OSX here is just remove the `GOOS ?= linux` line.

Comment (mandric) [reply Makefile]: Here is where it was added: https://github.com/mainflux/mainflux/pull/196

Comment (manuio) [reply Makefile]: Did you tested on Windows? I don't think that this is safe. Please, add again your previous version: 
`GOOS ?= $(shell go env GOOS)`
`GOARCH ?= $(shell go env GOARCH)`
`GOARM ?= $(shell go env GOARM)`

**Review by manuio — APPROVED** [review]

**Review by drasko — APPROVED** [review]
LGTM

**Comment by drasko** [comment]
Meged! @mandric welcome to Mainflux contributors club ;).

**Comment by mandric** [comment]
Awesome, thank you!

**Pull request merged**